### PR TITLE
Additional logs to view change process

### PR DESCRIPTION
**Pull request by HristoStaykov**
- printf-s replaced by LOG_DEBUG in ViewsManager.cpp
- added log when primary for next View sends NewView message to all replicas.
- added logs when a Replica has to wait for additional information before entering View.

### CODE REVIEW

File: bftengine/src/bftengine/ViewsManager.cpp [hunk 1]
```diff
@@ -568,13 +569,17 @@ bool ViewsManager::tryToEnterView(ViewNum v,
...
+  if (currentLastExecuted < currentLastStable) {
     // we don't have state, let's wait for state synchronization...
+    LOG_INFO(GL, "Waiting for state synchronization before entering view ...");
     return false;
+  }
 
-  if (currentLastStable < lowerBoundStableForPendingView)
+  if (currentLastStable < lowerBoundStableForPendingView) {
     // we don't have the latest stable point, let's wait for more information
+    LOG_INFO(GL, "Waiting for latest stable point before entering view ...");
```
Comment (teoparvanov) [inline bftengine/src/bftengine/ViewsManager.cpp:580]: Does it make sense to also log the specific view number you are referring to?

File: bftengine/src/bftengine/ViewsManager.cpp [hunk 2]
```diff
@@ -595,9 +600,11 @@ bool ViewsManager::tryToEnterView(ViewNum v,
 
     stat = Stat::PENDING;
 
-    if (currentLastStable < lowerBoundStableForPendingView)
+    if (currentLastStable < lowerBoundStableForPendingView) {
       // we don't have the latest stable point, let's wait for more information
+      LOG_INFO(GL, "New pending view. Waiting for latest stable point before entering view ...");
```
Comment (teoparvanov) [inline bftengine/src/bftengine/ViewsManager.cpp:605]: same here - which view?

File: bftengine/src/bftengine/ViewsManager.cpp [hunk 3]
```diff
@@ -736,7 +753,10 @@ bool ViewsManager::tryMoveToPendingViewAsPrimary(ViewNum v) {
     if (relatedVCMsgs.size() == SMAJOR) break;
   }
 
-  if (relatedVCMsgs.size() < SMAJOR) return false;
+  if (relatedVCMsgs.size() < SMAJOR) {
+    LOG_INFO(GL, "Waiting for sufficient ViewChange messages to entering view as primary ...");
```
Comment (teoparvanov) [inline bftengine/src/bftengine/ViewsManager.cpp:757]: Maybe for some of those log messages we can continue the tradition started by Guy - adding "*******" to highlight them in the logs? Probably not for all of the logs you've added - but at least the most important ones.

File: bftengine/src/bftengine/ViewsManager.cpp (hunk 1 above)
Comment (HristoStaykov) [reply bftengine/src/bftengine/ViewsManager.cpp:580]: Yes, definitely, will be easier for reviewing the log. Will add.

File: bftengine/src/bftengine/ViewsManager.cpp (hunk 3 above)
Comment (HristoStaykov) [reply bftengine/src/bftengine/ViewsManager.cpp:757]: I think it is good to have consistency in the logs for View change, will add the prefix. Thanks for the remark.

File: bftengine/src/bftengine/ReplicaImp.cpp [hunk 4]
```diff
@@ -1905,7 +1905,10 @@ void ReplicaImp::onNewView(const std::vector<PrePrepareMsg *> &prePreparesForNew
 
   // send messages
 
-  if (newNewViewMsgToSend != nullptr) sendToAllOtherReplicas(newNewViewMsgToSend);
+  if (newNewViewMsgToSend != nullptr) {
+    LOG_INFO(GL, "**************** Sending NewView message for view=" << curView << "to all replicas.");
```
Comment (teoparvanov) [inline bftengine/src/bftengine/ReplicaImp.cpp:1909]: nit: " to all replicas"

**Pull request merged**